src/snowflake/snowpark/dataframe.py

```diff
@@ -1358,7 +1358,7 @@ def select(
         for e in exprs:
             if isinstance(e, Column):
                 names.append(e._named())
-                if _emit_ast and ast:
+                if _emit_ast and ast and e._ast:
```

Comment:
Curious, why do you also need to check `e._ast` here?

Reply:
I added the check here since `make_interval` creates a Column object but it has no associated `_ast` field. Since `_emit_ast` is True by default, select tries to perform `ast.cols.append(e._ast)` and fails since `e._ast` is None (here `e` is the Column created by `make_interval`).

I don't think it is correct to do `_emit_ast=False` on the "user" side of things since it is supposed to be an interval parameter.

Reply:
oh, I think in that case you can remove the check for `e._ast` then. Simply assign in `make_interval` the `_ast` field. Trying to avoid here one-off solutions for an API. I think it should work without this (added) check. Can we try?

Reply:
removed the check, there was a bug in my test that made me run into this. Thanks for pointing it out!